subsys/dfu/Kconfig

```diff
@@ -14,6 +14,16 @@ menuconfig IMG_MANAGER
 
 if IMG_MANAGER
 
+config MCUBOOT_USE_SCRATCH
```

Comment:
Looks fine to me but I think worth to consider if no better assumed that if non-scratch algorithm is used, then DTS shouldn't contain scratch partition declaration at all (so for instance `DT_FLASH_AREA_IMAGE_SCRATCH_ID' label will not exist).
Alternativelly It might be good idea to disable this if dts doesn't contain scratch partition declaration.

Reply:
> Looks fine to me but I think worth to consider if no better assumed that if non-scratch algorithm is used, then DTS shouldn't contain scratch partition declaration at all (so for instance `DT_FLASH_AREA_IMAGE_SCRATCH_ID' label will not exist).  Alternativelly It might be good idea to disable this if dts doesn't contain scratch partition declaration.

Good idea, but I don't know how to implement this. Can we disable the `DT_` entries for scratch when non-scratch mode is enabled using `cmake` ? Or the other way around, can we check if the `DT_` entries do not exist and automatically enabled the non-scratch mode in `cmake`?

Reply:
DT scratch node can be removed using a DTS override file with `/node-delete/ &scratch_partition;` syntax inside. DTS Override file can be set via cmake. DT is compiled before kconfig, so kconfig might use DT values (zephyr has extension for that).

Reply:
I think I agree with this. This gives us potential to have MCUboot and Zephyr configured differently. At least basing it on the resulting DT makes more sense than a separate option.

It is important that an app and MCUboot end up with the same partition table, so there is still potential problems, but at least if this came from the resulting DT, it only has to be made consistent there.

Reply:
Why is this a Zephyr Kconfig option at all? It seems like this should be an mcuboot kconfig option only. We can make the code in Zephyr do the right thing based on the presence or absence of a scratch partition node in DT without an additional Kconfig option, no?